core/src/test/scala/org/broadinstitute/dsde/rawls/user/UserServiceSpec.scala

```diff
@@ -233,4 +236,169 @@ class UserServiceSpec extends AnyFlatSpecLike with TestDriverComponent with Mock
       actual.errorReport.statusCode shouldEqual Option(StatusCodes.Forbidden)
     }
   }
+
+  it should "set the spend configuration of a billing project when the user has permission" in {
+    withMinimalTestDatabase { dataSource: SlickDataSource =>
+      val billingProject = minimalTestData.billingProject
+      val spendReportDatasetName = "test_dataset"
+
+      val mockSamDAO = mock[SamDAO](RETURNS_SMART_NULLS)
+      when(mockSamDAO.userHasAction(SamResourceTypeNames.billingProject, billingProject.projectName.value, SamBillingProjectActions.alterSpendReportConfiguration, userInfo)).thenReturn(Future.successful(true))
+
+      val userService = getUserService(dataSource, mockSamDAO)
+
+      Await.result(userService.setProjectSpendConfiguration(billingProject.projectName, spendReportDatasetName), Duration.Inf) shouldEqual 1
+
+      val spendReportDatasetInDb = runAndWait(dataSource.dataAccess.rawlsBillingProjectQuery.filter(_.projectName === billingProject.projectName.value).map(_.spendReportDataset).result)
+
+      spendReportDatasetName shouldEqual spendReportDatasetInDb.head.get
+    }
+  }
+
```

Comment:
I think this case is covered with the existing tests where `userHasAction` returns `false`

They assert for 403 Forbidden because that's what the pre-existing `requireProjectAction` does. Perhaps that should be switched to a 404 at some point but it will consistently return 403 regardless of the reason (lack of action, or lack of overall access, resource does not exist at all) so I don't think it's leaky, at least.